pl_examples/domain_templates/generative_adversarial_net.py

```diff
@@ -241,7 +241,7 @@ def main(args: Namespace) -> None:
     # ------------------------
     # 1 INIT LIGHTNING MODEL
     # ------------------------
-    model = GAN(args)
+    model = GAN(lr=args.lr, b1=args.b1, b2=args.b2, latent_dim=args.latent_dim)
```

Comment:
```suggestion
    model = GAN(**vars(args))
```
can we simply do this? :]

Reply:
It will fail due to the too many arguments on GAN object. For this change, GAN class needs `**kwargs` as argument like the following, which I didn't do because it would harm the argument conciseness.

```
class GAN(LightningModule):
    ...
    def __init__(                                                                                                   
            self,                                                                                                   
            img_shape: tuple = (1, 28, 28),                                                                         
            lr: float = 0.0002,                                                                                     
            b1: float = 0.5,                                                                                        
            b2: float = 0.999,                                                                                      
            latent_dim: int = 100,                                                                                  
            **kwargs,                                                                                               
    ):                                                                                                              
...
```

 If you prefer this way, I will update the PR as you suggested.